### ISSUE TITLE
Unrecognized Flag '-Yu' in 'p2'

### ISSUE THREAD

**Issue by Tembocs** [microsoft/terminal#1031]
Trying to build from Windows 10 x64 using Visual Studio 2019 and got his error message.
```unrecognized flag '-Yu' in 'p2'```. This is from the current master branch. How do I resolve this? Is it something to do with UTF-8? What is the workaround then? Can't be applied to the whole repo? I have tried to search related issues without success with proposed solutions.

**Comment by zadjii-msft** [comment]
Could you give some more context on that error text? I've never seen it before.

**Comment by bjohnson-fsi** [comment]
Getting the same error when trying to build for x64 using Visual Studio 2019.

Full Error List:
`
Error	C1007	unrecognized flag '-Yu' in 'p2'	TerminalConnection	C:\Users\brinjo\Documents\GitHub\terminal\src\cascadia\TerminalConnection\LINK	1	
Error	LNK1257	code generation failed	TerminalConnection	C:\Users\brinjo\Documents\GitHub\terminal\src\cascadia\TerminalConnection\LINK	1	
Error	C2678	binary '&': no operator found which takes a left-hand operand of type 'const winrt::Windows::System::VirtualKeyModifiers' (or there is no acceptable conversion)	TerminalControl	c:\users\brinjo\documents\github\terminal\src\cascadia\terminalcontrol\termcontrol.cpp	695	
Error	C1001	An internal error has occurred in the compiler.	TerminalControl	c:\users\brinjo\documents\github\terminal\src\cascadia\terminalcontrol\termcontrol.cpp	695	
Error	MIDL2011	[msg]unresolved type declaration [context]: Microsoft.Terminal.Settings.KeyChord [ Parameter 'chord' of Procedure 'SetKeyBinding' ( RuntimeClass 'TerminalApp.AppKeyBindings' ) ]	TerminalApp	C:\Users\brinjo\Documents\GitHub\terminal\src\cascadia\TerminalApp\AppKeyBindings.idl	69	
Error	MIDL2011	[msg]unresolved type declaration [context]: Microsoft.Terminal.Settings.KeyChord [ Procedure 'GetKeyBinding' ( RuntimeClass 'TerminalApp.AppKeyBindings' ) ]	TerminalApp	C:\Users\brinjo\Documents\GitHub\terminal\src\cascadia\TerminalApp\AppKeyBindings.idl	70	
Error	MIDL9008	internal compiler problem - See documentation for suggestions on how to find a workaround.	TerminalApp	C:\Users\brinjo\Documents\GitHub\terminal\src\cascadia\TerminalApp\midl	1	
Error	C1083	 Cannot open include file: 'winrt/TerminalApp.h': No such file or directory	WindowsTerminal	C:\Users\brinjo\Documents\GitHub\terminal\src\cascadia\WindowsTerminal\AppHost.h	7	
Error	C1083	 Cannot open include file: 'winrt/TerminalApp.h': No such file or directory	WindowsTerminal	C:\Users\brinjo\Documents\GitHub\terminal\src\cascadia\WindowsTerminal\AppHost.h	7	
Error	C1083	 Cannot open include file: 'winrt/TerminalApp.h': No such file or directory	WindowsTerminal	C:\Users\brinjo\Documents\GitHub\terminal\src\cascadia\WindowsTerminal\IslandWindow.h	7	
Error	C1083	 Cannot open include file: 'winrt/TerminalApp.h': No such file or directory	WindowsTerminal	C:\Users\brinjo\Documents\GitHub\terminal\src\cascadia\WindowsTerminal\IslandWindow.h	7	
Error		This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is ..\..\..\packages\Microsoft.UI.Xaml.2.1.190405001-prerelease\build\native\Microsoft.UI.Xaml.targets.	CascadiaPackage	C:\Users\brinjo\Documents\GitHub\terminal\src\cascadia\CascadiaPackage\CascadiaPackage.wapproj	131	
`

**Comment by mmdanggg2** [comment]
I'm getting this too, trying to build x64 Release, Debug builds fine...
![image](https://user-images.githubusercontent.com/3161587/58750241-beea7a80-8487-11e9-923d-e56b63b42878.png)

**Comment by kopwei** [comment]
The same. Debug x64 build passed but Release x64 failed with same issue.

**Comment by baget** [comment]
> Could you give some more context on that error text? I've never seen it before.

I also saw this.

- Clone the repro, 
- Update the submodules 
- Opened with **VS2019**
- upgrade the solution
- build the solution.

This is the Error that I get
```
27>------ Build started: Project: Host.unittest, Configuration: Release x64 ------
25>readData.cpp
27>precomp.cpp
25>readDataDirect.cpp
25>readDataRaw.cpp
25>registry.cpp
15>Nihilist.vcxproj -> D:\MyProjects\GitHub\terminal\\bin\x64\Release\Nihilist.exe
24>LINK : fatal error C1007: unrecognized flag '-Yu' in 'p2'
24>LINK : fatal error LNK1257: code generation failed
```

**Comment by braindef** [comment]
i have the same Error...

**Comment by HimiSaber** [comment]
Right click  solution -->retarget solution，change to latest

**Comment by boldcronie** [comment]
I had the same error. VS2019/tools v142, and every build failed with at least two modules returning the "Unrecognized flag -Yu in p2" error. Re-targeted after clone, and individually re-targeted the UWP projects that didn't move from 141 to 142 during the initial re-target. Still got the errors during every build.

Tried HimiSaber's solution -- after another failed build, right-clicked the OpenConsole.sln and clicked Re-target solution. That finished, so I built again. No errors! Launching the WindowsTerminal.exe still didn't work (would appear, then disappear after a few seconds), but I deployed anyway. Only took a few seconds, as the solution had been properly built. Checked Start Menu, and Windows Terminal (Dev Build) was there and working!

Thank you HimiSaber! I almost didn't try it, as I'd already re-targeted several times.

**Comment by Tembocs** [comment]
@HimiSaber and @boldcronie, thank you so much this works. I have successful built WindowsTerminal and deploy. Its working and fast.

**Comment by DHowett-MSFT** [comment]
This will be fixed by #1012.

**Comment by NycuRO** [comment]
I still have this error.

**Comment by oliver-ni** [comment]
@NycuRO did you try @HimiSaber's solution? Worked for me

**Comment by NycuRO** [comment]
i fixed. but when i use ubuntuu profile i've got gray screen only.

**Comment by devinvisible** [comment]
I had a similar issue in a completely unrelated project. Going into my project properties and setting Linker -> General -> Show Progress to /VERBOSE gave me a clue -- it showed a library of mine that it was searching for. A stack overflow comment suggested this was caused by this library being built with a different version of visual studio. I rebuilt both using the same version of visual studio (being mindful to build the appropriate architecture & release) and this error went away.
